tools/project-context.js

```diff
@@ -985,7 +1013,7 @@ _.extend(exports.ProjectConstraintsFile.prototype, {
   eachConstraint: function (iterator) {
     var self = this;
     _.each(self._constraintLines, function (lineRecord) {
-      if (lineRecord.constraint)
+      if (! lineRecord.skipOnRead && lineRecord.constraint)
```

Comment:
`eachConstraint` iterates through `self._constraintLines` which may contain duplicates. This is because it needs to preserve all constraints from `.meteor/packages` (so this file can be later reproduced by `_write` method) as well as `self._includePackages`.

On the contrary, `getConstraint` reads from `self._constraintMap` which contains unique values.

Maybe there should be separate property for extra constraints instead of pushing them to `self._constraintLines`? For example `self._extraConstraintLines`. But then `eachConstraint` would need iterate through `self._extraConstraintLines` and `self._constraintLines` and find duplicates along the way.